blocks/test/full-content.js

```diff
@@ -152,6 +167,24 @@ describe( 'full post content fixture', () => {
 				}
 			}
 
+			// If file is suffixed as invalid, we expect a parse should fail.
+			const isExpectedValid = ! /(__|-)invalid$/.test( f );
+			const [ block ] = blocksActual;
+			if ( isExpectedValid !== block.isValid ) {
+				throw new Error( format(
+					isExpectedValid
+						? 'Parse failed on content from \'%s.html\':\n\n%s'
+						: 'Expected parse failure on content from \'%s.html\'',
+					f,
+					block.error || ''
+				) );
+			}
+			// Don't bother continuing with verifying or generating JSON and
+			// serialized HTML if we did not expect parse to succeed.
+			if ( ! isExpectedValid ) {
+				return;
+			}
+
```

Comment:
I think for an invalid parse result, we should still verify that the block is initialized and serialized as expected (currently, no change).